### ISSUE TITLE
🐛 [firebase_auth] Persistence is not working on web (tested in Chrome debug only)

### ISSUE THREAD

**Issue by Tom3652** [firebase/flutterfire#11717]
## Bug report

**Describe the bug**
The `User` is null after sign in and closing the window, and reopening when  `Persistence` is set to `LOCAL`.

### Steps to reproduce

Before doing the below steps, create a new email and password user account in your firebase console that you will use to test.

1. Run the sample code 
2. See the normal behaviour : 
- In logs : `User : null` 
- Then in logs `User : your_email` 
3. Kill the browser 
4. Comment the `FirebaseAuth.instance.signInWithEmailAndPassword` line in `main.dart`
5. Rebuild and see the logs `User : null`

### Expected behavior

At step 5, we should see the same behaviour as step 2 when the user should be found locally. 

### Sample project

```
import 'package:firebase_auth/firebase_auth.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
  await FirebaseAuth.instance.setPersistence(Persistence.LOCAL);
  FirebaseAuth.instance.userChanges().listen((event) {
    print("User : ${event?.email}");
  });
  await FirebaseAuth.instance.signInWithEmailAndPassword(email: "your_email", password: "your_password");

  runApp(const MaterialApp());
}
```

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 3.13.7, on macOS 14.0 23A344 darwin-x64, locale fr-FR)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.83.1)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 3.1.3
Flutter SDK 3.13.7
postonymous_website 1.0.0+1

dependencies:
- cupertino_icons 1.0.6
- firebase_auth 4.10.1 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 2.17.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math web sky_engine]

dev dependencies:
- flutter_lints 2.0.3 [lints]
- flutter_test 0.0.0 [flutter test_api matcher path fake_async clock stack_trace vector_math async boolean_selector characters collection material_color_utilities meta source_span stream_channel string_scanner term_glyph web]

transitive dependencies:
- _flutterfire_internals 1.3.7 [collection firebase_core firebase_core_platform_interface flutter meta]
- async 2.11.0 [collection meta]
- boolean_selector 2.1.1 [source_span string_scanner]
- characters 1.3.0
- clock 1.1.1
- collection 1.17.2
- fake_async 1.3.1 [clock collection]
- firebase_auth_platform_interface 6.19.1 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 5.8.2 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser js meta]
- firebase_core_platform_interface 4.8.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.8.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- flutter_web_plugins 0.0.0 [flutter characters collection material_color_utilities meta vector_math web]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- js 0.6.7 [meta]
- lints 2.1.1
- matcher 0.12.16 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.5.0 [collection]
- meta 1.9.1
- path 1.8.3
- plugin_platform_interface 2.1.6 [meta]
- sky_engine 0.0.99
- source_span 1.10.0 [collection path term_glyph]
- stack_trace 1.11.0 [path]
- stream_channel 2.1.1 [async]
- string_scanner 1.2.0 [source_span]
- term_glyph 1.2.1
- test_api 0.6.0 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph]
- typed_data 1.3.2 [collection]
- vector_math 2.1.4
- web 0.1.4-beta
```

</details>

---

**Comment by drklrd56** [comment]
I also faced this issue.. The way around is adding this after firebase app initialization. 
```
FirebaseAuth.instance
          .authStateChanges()
          .distinct((first, second) => first?.uid == second?.uid);
```
This resolves the auth issue. I haven't faced this issue for 6 months now. But again this is a way around not a solution

**Comment by Tom3652** [comment]
Thanks for the suggestion, but this is still not working. 

I don't see how your workaround for the current issue could even work since it's only a filter on the `authStateChanges` result, which is in that case only null after the app restart

**Comment by darshankawar** [comment]
Thanks for the report.
I was able to replicate the reported behavior.

```
Performing hot restart...                                          254ms
Restarted application in 256ms.
The platformViewRegistry getter is deprecated and will be removed in a future release. Please import it from `dart:ui_web` instead.
User : null
User : testdan248@gmail.com
User : testdan248@gmail.com
```

Then

```
The platformViewRegistry getter is deprecated and will be removed in a future release. Please import it from `dart:ui_web` instead.
The Flutter DevTools debugger and profiler on Chrome is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:64913/ePhBlOJnlqc=
User : null
```

**Comment by Lyokone** [comment]
Hey 👋 
In Chrome Debug it's expected since each restart of the Chrome Debug browser, the browser is starting from a clean state. Also to have the state to be persisted locally you need to keep the same port between restarts.
